TypeCobol/Compiler/Preprocessor/CopyTokensLinesIterator.cs

```diff
@@ -50,17 +50,18 @@ private struct CopyTokensLinesIteratorPosition
 
         // Previous snapshot position
         private object snapshotPosition;
-        private bool AllowWhitespaceTokens;
+        // Allow the caller to Filter some tokens.
+        private Func<Token, bool> TokenFilterCallback;
 
         /// <summary>
         /// Set the initial position of the iterator with startToken.
         /// </summary>
-        public CopyTokensLinesIterator(string sourceFileName, IReadOnlyList<IProcessedTokensLine> tokensLines, int channelFilter, bool allowWhitespaceTokens)
+        public CopyTokensLinesIterator(string sourceFileName, IReadOnlyList<IProcessedTokensLine> tokensLines, int channelFilter, Func<Token, bool> tokenFilterCallback)
```

Comment:
Both `channelFilter` and `tokenFilterCallback` are used to select/discard tokens during iteration so they should be merged together, adding a default value (filtering based on channel like before the change).